jxl/src/render/low_memory_pipeline/save/mod.rs

```diff
@@ -136,6 +156,50 @@ impl SaveStage {
                 }
             }
         }
+
+        // Fill opaque alpha if needed (when RGBA requested but image has no alpha)
```

Comment:
I believe this would be better implemented as follows:
- Let's add additional `RowBuffer`s to the LowMemoryRenderPipeline (per save stage), with one row of storage each, which are only populated if fill_opaque_alpha is true.
- Those `RowBuffer`s are populated with a fully-opaque alpha value *for the correct bitdepth* (now we'd use `65535u16` even if requesting 10-bit output, which is incorrect).
- Then, we can just append a reference to that RowBuffer in the vector we pass as `data`; this removes the need for handling `fill_opaque_alpha` in any special way, and so should get rid of ~all the changes to this submodule.

We should follow a similar approach also in the `simple_render_pipeline`, and add tests that compare that filling opaque alpha has the same behaviour with both pipelines.